File: app/src/main/java/com/gabra/deliverybot/DeliveryBotService.java
package com.gabra.deliverybot;

import android.accessibilityservice.AccessibilityService;
import android.accessibilityservice.AccessibilityServiceInfo;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.accessibility.AccessibilityEvent;
import android.view.accessibility.AccessibilityNodeInfo;

import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class DeliveryBotService extends AccessibilityService {
    private static final String TAG = "DeliveryBotService";
    private boolean isServiceEnabled = false;
    private boolean isTestMode = false;
    private SharedPreferences sharedPreferences;

    @Override
    public void onServiceConnected() {
        super.onServiceConnected();
        Log.d(TAG, "Service connected");
        
        AccessibilityServiceInfo info = new AccessibilityServiceInfo();
        info.eventTypes = AccessibilityEvent.TYPES_ALL_MASK;
        info.feedbackType = AccessibilityServiceInfo.FEEDBACK_GENERIC;
        info.notificationTimeout = 100;
        setServiceInfo(info);
        
        isServiceEnabled = true;
        Intent intent = new Intent(ACTION_UPDATE_STATUS);
        intent.putExtra("status", true);
        sendBroadcast(intent);
    }

    @Override
    public void onAccessibilityEvent(AccessibilityEvent event) {
        if (!isServiceEnabled) return;

        if (event.getEventType() == AccessibilityEvent.TYPE_WINDOW_STATE_CHANGED) {
            processWindowEvent(event);
        } else if (event.getEventType() == AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED) {
            processContentEvent(event);
        }
    }

    private void processWindowEvent(AccessibilityEvent event) {
        if (event.getPackageName() == null) return;
        
        if (event.getPackageName().toString().contains("com.mistermandob")) {
            Log.d(TAG, "Mister Mandob window detected");
            processDeliveryRequest();
        }
    }

    private void processContentEvent(AccessibilityEvent event) {
        if (event.getPackageName() == null) return;
        
        if (event.getPackageName().toString().contains("com.mistermandob")) {
            Log.d(TAG, "Content changed in Mister Mandob");
            processDeliveryRequest();
        }
    }

    private void processDeliveryRequest() {
        AccessibilityNodeInfo root = getRootInActiveWindow();
        if (root == null) return;

        // Find distance node
        List<AccessibilityNodeInfo> distanceNodes = root.findAccessibilityNodeInfosByText("كم");
        if (distanceNodes.isEmpty()) {
            Log.d(TAG, "No distance found");
            return;
        }

        // Extract distance
        String distanceText = distanceNodes.get(0).getText().toString();
        Pattern pattern = Pattern.compile("(\\d+(?:\\.\\d+)?) كم");
        Matcher matcher = pattern.matcher(distanceText);
        if (!matcher.find()) {
            Log.d(TAG, "Invalid distance format");
            return;
        }

        double distance = Double.parseDouble(matcher.group(1));
        Log.d(TAG, "Distance: " + distance + " km");

        // Check if distance is acceptable
        if (distance <= 8.0) {
            // Find and click "تفاصيل" button
            List<AccessibilityNodeInfo> detailsButtons = root.findAccessibilityNodeInfosByText("تفاصيل");
            if (!detailsButtons.isEmpty()) {
                AccessibilityNodeInfo detailsButton = detailsButtons.get(0);
                if (detailsButton.isClickable()) {
                    detailsButton.performAction(AccessibilityNodeInfo.ACTION_CLICK);
                    Log.d(TAG, "Clicked تفاصيل");

                    // Wait and scroll
                    try {
                        Thread.sleep(1000);
                        root.performAction(AccessibilityNodeInfo.ACTION_SCROLL_FORWARD);
                        Log.d(TAG, "Scrolled");

                        // Find and click "قبول" button
                        List<AccessibilityNodeInfo> acceptButtons = root.findAccessibilityNodeInfosByText("قبول");
                        if (!acceptButtons.isEmpty()) {
                            AccessibilityNodeInfo acceptButton = acceptButtons.get(0);
                            if (acceptButton.isClickable()) {
                                if (!isTestMode) {
                                    acceptButton.performAction(AccessibilityNodeInfo.ACTION_CLICK);
                                    Log.d(TAG, "Accepted order");
                                } else {
                                    Log.d(TAG, "Test mode: Would have accepted order");
                                }
                            }
                        }
                    } catch (InterruptedException e) {
                        Log.e(TAG, "Error during processing", e);
                    }
                }
            }
        }
    }

    @Override
    public void onInterrupt() {
        Log.d(TAG, "Service interrupted");
        isServiceEnabled = false;
        Intent intent = new Intent(ACTION_UPDATE_STATUS);
        intent.putExtra("status", false);
        sendBroadcast(intent);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "Service destroyed");
        isServiceEnabled = false;
        Intent intent = new Intent(ACTION_UPDATE_STATUS);
        intent.putExtra("status", false);
        sendBroadcast(intent);
    }

    private static final String ACTION_UPDATE_STATUS = "com.gabra.deliverybot.UPDATE_STATUS";
}

File: app/src/main/java/com/gabra/deliverybot/DeliveryBotService.kt
package com.gabra.deliverybot

import android.accessibilityservice.AccessibilityService
import android.accessibilityservice.GestureResultCallback
import android.accessibilityservice.GestureDescription
import android.content.Context
import android.graphics.Path
import android.os.Build
import android.view.accessibility.AccessibilityEvent
import android.view.accessibility.AccessibilityNodeInfo
import androidx.annotation.RequiresApi
import timber.log.Timber

class DeliveryBotService : AccessibilityService() {
    private var isServiceEnabled = false
    private var isTestingMode = false

    @RequiresApi(Build.VERSION_CODES.N)
    override fun onAccessibilityEvent(event: AccessibilityEvent?) {
        if (!isServiceEnabled) return

        event?.let { processEvent(it) }
    }

    private fun processEvent(event: AccessibilityEvent) {
        val rootNode = rootInActiveWindow ?: return
        
        // Check for new delivery request
        val distanceNode = findDistanceNode(rootNode)
        if (distanceNode != null) {
            val distance = extractDistance(distanceNode.text.toString())
            if (distance < 8.0) {
                handleOrder(distance)
            }
        }
    }

    private fun findDistanceNode(node: AccessibilityNodeInfo): AccessibilityNodeInfo? {
        // Implementation for finding distance node
        return null
    }

    private fun extractDistance(text: String): Double {
        // Implementation for extracting distance from text
        return 0.0
    }

    private fun handleOrder(distance: Double) {
        Timber.d("Handling order with distance: $distance km")
        
        val rootNode = rootInActiveWindow ?: return
        
        // Click Details button
        val detailsButton = findDetailsButton(rootNode)
        if (detailsButton != null) {
            performClick(detailsButton)
            
            // Scroll to accept button
            val acceptButton = findAcceptButton(rootNode)
            if (acceptButton != null) {
                performScrollTo(acceptButton)
                
                if (!isTestingMode) {
                    performClick(acceptButton)
                }
            }
        }
    }

    private fun findDetailsButton(node: AccessibilityNodeInfo): AccessibilityNodeInfo? {
        // Implementation for finding details button
        return null
    }

    private fun findAcceptButton(node: AccessibilityNodeInfo): AccessibilityNodeInfo? {
        // Implementation for finding accept button
        return null
    }

    private fun performClick(node: AccessibilityNodeInfo) {
        node.performAction(AccessibilityNodeInfo.ACTION_CLICK)
    }

    private fun performScrollTo(targetNode: AccessibilityNodeInfo) {
        val path = Path()
        path.moveTo(0f, 0f)
        path.lineTo(0f, 500f)
        
        val gestureBuilder = GestureDescription.Builder()
        gestureBuilder.addStroke(GestureDescription.StrokeDescription(path, 0, 500))
        
        dispatchGesture(gestureBuilder.build(), object : GestureResultCallback() {
            override fun onCompleted(gestureDescription: GestureDescription?) {
                super.onCompleted(gestureDescription)
                Timber.d("Scroll completed")
            }
            
            override fun onCancelled(gestureDescription: GestureDescription?) {
                super.onCancelled(gestureDescription)
                Timber.d("Scroll cancelled")
            }
        }, null)
    }

    override fun onInterrupt() {
        Timber.d("Service interrupted")
    }

    override fun onServiceConnected() {
        super.onServiceConnected()
        Timber.d("Service connected")
    }
}
